.github/workflows/openapi.yml

```diff
@@ -0,0 +1,53 @@
+name: openapi
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  deref_json:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Getsentry Token
+      id: getsentry
+      uses: getsentry/action-github-app-token@v1
+      with:
+        app_id: ${{ secrets.SENTRY_INTERNAL_APP_ID }}
+        private_key: ${{ secrets.SENTRY_INTERNAL_APP_PRIVATE_KEY }}
+
+    - name: Checkout getsentry/sentry
+      uses: actions/checkout@v2
+      with:
+        path: sentry
+
+    - name: Checkout getsentry/sentry-api-schema
+      uses: actions/checkout@v2
+      with:
+        ref: 'main'
+        repository: getsentry/sentry-api-schema
+        path: sentry-api-schema
+        token: ${{ steps.getsentry.outputs.token }}
+
+    # Install/setup node
+    - uses: volta-cli/action@v1
+
+    - name: Build OpenAPI Derefed JSON
+      run: |
+        cd sentry
+        yarn install --frozen-lockfile
+        yarn run build-derefed-docs
```

Comment:
@NisanthanNanthakumar once I merge https://github.com/getsentry/sentry/pull/20901 you'll have to specify the path, since the default will now be the tests path. so basically you'll need 

```
yarn run build-derefed-docs sentry/api-docs/openapi-derefed.json
```